runtime/src/main/java/org/corfudb/security/tls/ReloadableTrustManager.java

```diff
@@ -0,0 +1,148 @@
+package org.corfudb.security.tls;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This trust manager reloads the trust store upon certificate verification failure.
+ *
+ * Created by zjohnny on 9/18/17.
+ */
+@Slf4j
+public class ReloadableTrustManager implements X509TrustManager {
+    private String trustStorePath, trustPasswordPath;
+    private X509TrustManager trustManager;
+
+    /**
+     * Constructor.
+     *
+     * @param trustStorePath
+     *          Location of trust store.
+     * @param trustPasswordPath
+     *          Location of trust store password.
+     * @throws SSLException
+     *          Thrown when there's an issue with loading the trust store.
+     */
+    public ReloadableTrustManager(String trustStorePath, String trustPasswordPath) throws SSLException {
+        this.trustStorePath = trustStorePath;
+        this.trustPasswordPath = trustPasswordPath;
+        reloadTrustStore();
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        try {
+            reloadTrustStore();
+        } catch (SSLException e) {
+            log.error("Unable to reload trust store.", e);
```

Comment:
Should throw a CertificateException after the log here.
If not the subsequent checkClientTrusted() can potentially make a decision based on old trust store.